lib/java/src/org/apache/thrift/partial/ThriftMetadata.java

```diff
@@ -463,14 +463,12 @@ protected void toPrettyString(StringBuilder sb, int level) {
     }
 
     public <T extends TBase> T createNewStruct() {
-      T instance = null;
+      T instance;
 
       try {
         Class<T> structClass = getStructClass(this.data);
-        instance = (T) structClass.newInstance();
-      } catch (InstantiationException e) {
-        throw new RuntimeException(e);
-      } catch (IllegalAccessException e) {
+        instance = structClass.newInstance();
```

Comment:
Should this use `.getDeclaredConstructor().newInstance()`? Or is that too new for Thrift's minimum Java version?

Reply:
fixed